connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -398,15 +390,19 @@ private <T> Collection<PluginDesc<T>> getPluginDesc(
         }
 
         Collection<PluginDesc<T>> result = new ArrayList<>();
-        for (Class<? extends T> plugin : plugins) {
-            if (PluginUtils.isConcrete(plugin)) {
-                try {
-                    result.add(pluginDesc(plugin, versionFor(plugin), loader));
-                } catch (ReflectiveOperationException | LinkageError e) {
-                    log.error("Failed to discover {}: Unable to instantiate {}{}", klass.getSimpleName(), plugin.getSimpleName(), reflectiveErrorDescription(e), e);
-                }
-            } else {
-                log.debug("Skipping {} as it is not concrete implementation", plugin);
+        for (Class<? extends T> pluginKlass : plugins) {
+            if (!PluginUtils.isConcrete(pluginKlass)) {
+                log.debug("Skipping {} as it is not concrete implementation", pluginKlass);
+                continue;
+            }
+            if (!isParentClassloader(pluginKlass.getClassLoader(), loader)) {
+                log.debug("Exclude {} that is from classloader {}", pluginKlass.getSimpleName(), pluginKlass.getClassLoader());
```

Comment:
Should we also state why this classloader warrants exclusion of the scanned plugin?